src/Model/ElementForm.php

```diff
@@ -44,6 +47,11 @@ public function Form()
         $link = $current->Link();
         restore_error_handler();
 
+        if ($isFinishedAction) {
+            $pattern = '#element/\d+#'; // Regular expression pattern for handleElement (element/$ID!)
+            $link = preg_replace($pattern, '', $link);
+        }
```

Comment:
While this works in this specific scenario, it feels a bit flimsy relying on hardcoded URL segments.
This could fall apart if, for example, the elemental userform is on a page (probably with a custom controller) which has the url `/some-page/element/1/` - the link for the userform element then would end up being `/some-page/element/1/element/<id>/` and the regex would match and remove both sets of element and ID.

One solution would be to make the following change in `ElementFormController`:
```diff
 public function Link($action = null)
 {
-    $id = $this->element->ID;
-    $segment = Controller::join_links('element', $id, $action);
     $page = Director::get_current_page();
+    $id = $this->element->ID;
+    if ($this->getAction() === 'finished'
+        && $action !== 'finished'
+        && $this->getRequest()->param('ID') == $id
+    ) {
+        $segment = $action;
+    } else {
+        $segment = Controller::join_links('element', $id, $action);
+    }
 
     if ($page && !($page instanceof ElementController)) {
```

This avoids adding the `element/<id>/` portion of the URL if:
1. We are handling a "finished" request
2. The URL we're fetching is not for an action called "finished" (could also swap this out for `($action !== '' && $action !== null)`)
3. The ID the "finished" request is for is the current element

I feel like even that might have some edge cases I've not considered though...

Reply:
This solution is much better than mine. I implemented the changes.
Thanks :-)